airbyte-integrations/connectors/source-harvest/source_harvest/manifest.yaml

```diff
@@ -0,0 +1,2832 @@
+    name: expenses_categories
+    primary_key: []
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://api.harvestapp.com/v2/
+        path: /reports/expenses/categories
+        http_method: GET
+        request_parameters:
+          per_page: "50"
+        request_headers:
+          Harvest-Account-Id: "{{ config['account_id'] }}"
+        authenticator:
+          type: SelectiveAuthenticator
+          authenticator_selection_path: ["auth_type"]
+          authenticators:
+            api_token:
+              type: BearerAuthenticator
+              api_token: "{{ config['api_token'] }}"
+            oauth:
+              type: OAuthAuthenticator
+              client_id: "{{ config['client_id'] }}"
+              grant_type: refresh_token
+              client_secret: "{{ config['client_secret'] }}"
+              refresh_token: "{{ config['client_refresh_token'] }}"
+              refresh_request_body: {}
+              token_refresh_endpoint: https://id.getharvest.com/api/v2/oauth2/token
+        error_handler:
+          type: CompositeErrorHandler
+          error_handlers:
+            - type: DefaultErrorHandler
+              backoff_strategies:
+                - type: WaitTimeFromHeader
+                  header: Retry-After
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - results
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 1
+          cursor_value: '{{ response.get("links", {}).get("next", {}) }}'
+          stop_condition: '{{ not response.get("links", {}).get("next", {}) }}'
+      partition_router: []
+    transformations:
+      - type: AddFields
+        fields:
+          - path:
+              - from
+            value: '"{{ stream_partition.start_time }}"'
+      - type: AddFields
+        fields:
+          - path:
+              - to
+            value: '"{{ stream_partition.end_time }}"'
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: to
+      cursor_datetime_formats:
+        - "%Y%m%d"
+      datetime_format: "%Y%m%d"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+      start_time_option:
+        type: RequestOption
+        field_name: from
+        inject_into: request_parameter
+      end_time_option:
+        type: RequestOption
+        field_name: to
+        inject_into: request_parameter
+      end_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ now_utc().strftime('%Y-%m-%dT%H:%M:%SZ') }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+      step: P365D
+      cursor_granularity: P1D
+  - type: DeclarativeStream
```

Comment:
Differences observed:
* The previous implementation would use different boundaries `{"from": "20210101", "to": "20220101"}` instead of `{"from": "20210101", "to": "20211231"}`. As [Harvest query params include the boundaries](https://help.getharvest.com/api-v2/reports-api/reports/expense-reports/#clients-report), it seems fine but don't hesitate to do more testing around that

Reply:
Fair point, I'll add testing to verify my assumptions.